src/ln/channel.rs

```diff
@@ -229,6 +229,22 @@ enum ChannelState {
 }
 const BOTH_SIDES_SHUTDOWN_MASK: u32 = (ChannelState::LocalShutdownSent as u32 | ChannelState::RemoteShutdownSent as u32);
 
+// TODO: MAY fail the channel if
+// funding_satoshi is too small
+// htlc_minimum_msat too large
+// max_htlc_value_in_flight_msat too small
+// channel_reserve_satoshis too large
+// max_accepted_htlcs too small
+// dust_limit_satoshis too small
+struct ChannelConstraints {
```

Comment:
This appears unused?

Reply:
I wanted to leave this placeholder for the future. I had no clear idea whether this info needs to come from the user or decided internally (but how) and also which structure should hold this.

Reply:
A better way might be to add a comment that lists all the possible restrictions instead of adding an unused struct which adds (more) compile-time warnings.